File: DataStructure/Queue/SimpleQueue.java

public class SimpleQueue extends SimpleList {

	public SimpleQueue() {
		super();
	}
	
	public Object dequeue(){
		if(head.getNext()!=tail){			
			return super.remove(0); // Remove the first element
		}
		return null;
	}
	
	public int enqueue(Object obj){
		return super.add(obj); // add object to the last index
	}
	
	 @Override
	public int add(Object obj) {
		return enqueue(obj);
	}
	 
	 @Override
	public Object remove(int index) {
		return dequeue();
	}

}

File: DataStructure/Queue/Python/SimpleQueue.py
from SimpleList import SimpleList


class SimpleQueue(SimpleList):
    def __init__(self):
        SimpleList.__init__(self)

    def dequeue(self):
        if self.head.next is not self.tail:
            return SimpleList.remove(self, 0)
        return None

    def enqueue(self, obj):
        return SimpleList.add(self, obj)

    def add(self, obj):
        return self.enqueue(obj)

    def remove(self, index):
        return self.dequeue()